Makefile

```diff
@@ -134,13 +134,19 @@ dist/package.json : src/js/package.json
 npm-link: dist/package.json
 	cd src/test-js && npm ci && npm link ../../dist
 
-dist/pyodide.d.ts dist/pyodide/ffi.d.ts: src/js/*.ts src/js/pyproxy.gen.ts src/js/error_handling.gen.ts
-	npx dts-bundle-generator src/js/{pyodide,ffi}.ts --export-referenced-types false --project src/js/tsconfig.json
+dist/pyodide.d.ts dist/pyodide/ffi.d.ts: src/js/*.ts src/js/pyproxy.gen.ts src/js/error_handling.gen.ts src/js/generated_struct_info32.gen.json
+	node src/js/esbuild.config.mjs
+	# See https://github.com/timocov/dts-bundle-generator/issues/255
+	echo "export declare const defines: {[k:string] : number};" > src/js/generated_struct_info32.gen.json.d.ts
+	cd src/js && npx dts-bundle-generator {pyodide,ffi}.ts --export-referenced-types false --project tsconfig.json
 	mv src/js/{pyodide,ffi}.d.ts dist
 	python3 tools/fixup-type-definitions.py dist/pyodide.d.ts
 	python3 tools/fixup-type-definitions.py dist/ffi.d.ts
 
 
+src/js/generated_struct_info32.gen.json : emsdk/emsdk/.complete
+	cp emsdk/emsdk/upstream/emscripten/src/generated_struct_info32.json $@
```

Comment:
How about embedding this JSON file (or maybe a TS-converted version of it) into `/src/js` and updating it when we update emscripten version (we will definitely need to automate it).

Looking at this PR, I think things getting a bit complex just to handle this file. I am even okay with hardcoding constants that we use if it doesn't change often.

Reply:
Fine with me. I'd rather use this file, since if they do change it would be pretty hard to figure out why things were breaking.

Reply:
Okay, I checked in this file and added a check to `emsdk/Makefile` to check that the files match after installing emscripten.